espnet/nets/pytorch_backend/e2e_asr_transformer.py

```diff
@@ -160,6 +162,44 @@ def reset_parameters(self, args):
         # initialize parameters
         initialize(self, args.transformer_init)
 
+    def wfst_based_decoding(self, x, HLG, recog_args, char_list=None):
+        """wfst-based decoding.
+
+        :param ndnarray x: input acoustic feature (B, T, D) or (T, D)
+        :param Namespace recog_args: argment Namespace contraining options
+        :param list char_list: list of characters
+        :return: decoding result
+        :rtype: list
+        """
+        self.eval()
+        encoder_out, encoder_mask = self.encoder(x, None)
+        log_probs = self.ctc.log_softmax(
+            encoder_out)
+        log_probs[:, :, 0] += -1
+        dense_fsa_vec_seg = torch.stack(
+             (torch.tensor(range(encoder_out_lens.shape[0])),
+             torch.zeros(encoder_out_lens.shape[0]),
+             encoder_out_lens.to("cpu")), 1).to(torch.int32)
+        indices = torch.argsort(dense_fsa_vec_seg[:, 2], descending=True)
+        dense_fsa_vec_seg = dense_fsa_vec_seg[indices]
+        dense_fsa_vec = k2.DenseFsaVec(log_probs, dense_fsa_vec_seg)
+        lattices = k2.intersect_dense_pruned(HLG, dense_fsa_vec, 30.0, 7.0, recog_args.beam_size, 10000)
```

Comment:
According to the documentation of `k2.intersect_dense_pruned`
https://k2.readthedocs.io/en/latest/python_api/fsa_algo.html#k2.intersect_dense_pruned

The position where 7.0 occupies is the `beam_size` (i.e., `output_beam`). The current position for
`recog_args.beam_size` is actually `min_active_states`.